mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.tmpl

```diff
@@ -6080,3 +6080,197 @@ resource "google_container_node_pool" "np" {
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerNodePool_acceleratorNetworkProfile(clusterName, npName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", npName),
+					resource.TestCheckResourceAttr(resourceName, "node_config.0.machine_type", "a3-highgpu-8g"),
+					resource.TestCheckResourceAttr(resourceName, "network_config.0.accelerator_network_profile", "auto"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{"deletion_protection","cluster"},
+			},
+		},
+	})
+}
+
+func testAccContainerNodePool_acceleratorNetworkProfile_conflict(clusterName, npName string) string {
+	return fmt.Sprintf(`
+// 1. Create the "additional" network
+resource "google_compute_network" "additional_net" {
+  name                    = "%[1]s-net" // Based on clusterName
+  auto_create_subnetworks = false
+}
+
+// 2. Create the "additional" subnetwork
+resource "google_compute_subnetwork" "additional_sub" {
+  name          = "%[1]s-sub" // Based on clusterName
+  network       = google_compute_network.additional_net.id
+  ip_cidr_range = "10.2.0.0/16"
+  region        = "us-central1"
+}
+
+// 3. Create the cluster (this will use the 'default' network)
+resource "google_container_cluster" "cluster" {
+  name     = "%[1]s" // clusterName
+  location = "us-central1-a"
+
+  datapath_provider       = "ADVANCED_DATAPATH"
+  ip_allocation_policy    {} // Enables IP aliasing
+  enable_multi_networking = true
+
+  initial_node_count = 1
+  deletion_protection = false
+}
+
+// 4. Create the node pool with the conflicting config
+resource "google_container_node_pool" "np" {
+  name     = "%[2]s" // npName
+  cluster  = google_container_cluster.cluster.id
+  location = "us-central1-a"
+  node_count = 1
+
+  node_config {
+    machine_type = "a3-highgpu-8g"
+  }
+
+  network_config {
+    // --- This is the conflict ---
+    accelerator_network_profile = "auto"
+
+    additional_node_network_configs {
+      // Use the REAL network and subnetwork IDs
+      network    = google_compute_network.additional_net.id
+      subnetwork = google_compute_subnetwork.additional_sub.name 
+    }
+  }
+}
+`, clusterName, npName)
+}
+
+func TestAccContainerNodePool_acceleratorNetworkProfile_conflict(t *testing.T) {
+    t.Parallel()
+
+    clusterName := fmt.Sprintf("tf-test-cluster-%s", acctest.RandString(t, 10))
+    npName := fmt.Sprintf("tf-test-nodepool-%s", acctest.RandString(t, 10))
+
+    resource.Test(t, resource.TestCase{
+        PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+        ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+        // No CheckDestroy is needed because the resource should never be created
```

Comment:
Since this is testing google_container_cluster, a CheckDestroy would be appropriate.